From 63638aee04429b7fd2f937a3005f96187082b8c1 Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date: Fri, 22 Nov 2013 21:29:58 +0100
Subject: [PATCH 26/42] winepulse: disable the setevent part of the latency
 hack

If you get playback glitches in skyrim or other games as a result of
this patch, PLEASE REPORT TO ME!
---
 dlls/winepulse.drv/mmdevdrv.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/dlls/winepulse.drv/mmdevdrv.c b/dlls/winepulse.drv/mmdevdrv.c
index 68de00c..643d55e 100644
--- a/dlls/winepulse.drv/mmdevdrv.c
+++ b/dlls/winepulse.drv/mmdevdrv.c
@@ -1822,7 +1822,7 @@ static HRESULT WINAPI AudioRenderClient_ReleaseBuffer(
 {
     ACImpl *This = impl_from_IAudioRenderClient(iface);
     UINT32 written_bytes = written_frames * pa_frame_size(&This->ss);
-    UINT32 period;
+//    UINT32 period;
 
     TRACE("(%p)->(%u, %x)\n", This, written_frames, flags);
 
@@ -1858,10 +1858,10 @@ static HRESULT WINAPI AudioRenderClient_ReleaseBuffer(
     TRACE("Released %u, pad %zu\n", written_frames, This->pad / pa_frame_size(&This->ss));
     assert(This->pad <= This->bufsize_bytes);
 
-    period = pa_stream_get_buffer_attr(This->stream)->minreq;
+//    period = pa_stream_get_buffer_attr(This->stream)->minreq;
     /* Require a minimum of 3 periods filled, if possible */
-    if (This->event && This->pad + period <= This->bufsize_bytes && This->pad < period * 3)
-        SetEvent(This->event);
+//    if (This->event && This->pad + period <= This->bufsize_bytes && This->pad < period * 3)
+//        SetEvent(This->event);
     pthread_mutex_unlock(&pulse_lock);
     return S_OK;
 }
-- 
1.8.4.4

